File: frontend/src/api/identity.ts
import { apiGet, apiPatch, apiPost } from './client'

export interface Workspace {
  id: string
  name: string
  slug: string
  deployment_mode: string
  default_locale: string
  timezone: string
  department_max_depth: number
  status: string
}

export interface User {
  id: string
  display_name: string
  email?: string | null
  avatar_url?: string | null
  status: string
}

export interface Member {
  id: string
  workspace_id: string
  user_id: string
  department_id?: string | null
  display_name: string
  email?: string | null
  employee_no?: string | null
  avatar_url?: string | null
  status: string
}

export interface CurrentPrincipal {
  user: User
  member: Member
  workspace: Workspace
  permissions: string[]
  roles: string[]
}

export interface SecuritySettings {
  workspace_admin_can_view_all_submitted_reports: boolean
  audit_enabled: boolean
  private_deployment_single_workspace: boolean
}

export interface AuthProviderConfig {
  id: string
  workspace_id: string
  provider_type: string
  name: string
  enabled: boolean
  config_public: Record<string, unknown>
  bind_password_configured: boolean
}

export interface AuthProviderConfigUpdate {
  name?: string
  enabled?: boolean
  config_public?: Record<string, unknown>
  bind_password?: string
}

export interface AuthProviderTestResult {
  ok: boolean
  status: string
  message: string
}

export function getCurrentPrincipal() {
  return apiGet<CurrentPrincipal>('/auth/me')
}

export function getWorkspace() {
  return apiGet<Workspace>('/workspace')
}

export function listMembers() {
  return apiGet<Member[]>('/members')
}

export function getSecuritySettings() {
  return apiGet<SecuritySettings>('/workspace/security-settings')
}

export function getLdapProviderConfig() {
  return apiGet<AuthProviderConfig>('/auth/providers/ldap')
}

export function updateLdapProviderConfig(payload: AuthProviderConfigUpdate) {
  return apiPatch<AuthProviderConfig, AuthProviderConfigUpdate>('/auth/providers/ldap', payload)
}

export function testLdapProviderConfig() {
  return apiPost<AuthProviderTestResult, Record<string, never>>('/auth/providers/ldap/test', {})
}

File: backend/app/schemas/identity.py
from uuid import UUID

from pydantic import BaseModel, Field

from app.schemas.common import OrmModel


class WorkspaceRead(OrmModel):
    id: UUID | str
    name: str
    slug: str
    deployment_mode: str
    default_locale: str
    timezone: str
    department_max_depth: int
    status: str


class WorkspaceUpdate(BaseModel):
    name: str | None = Field(default=None, min_length=1, max_length=120)
    timezone: str | None = Field(default=None, max_length=64)
    department_max_depth: int | None = Field(default=None, ge=1, le=8)


class UserRead(OrmModel):
    id: UUID | str
    display_name: str
    email: str | None = None
    avatar_url: str | None = None
    status: str


class MemberRead(OrmModel):
    id: UUID | str
    workspace_id: UUID | str
    user_id: UUID | str
    department_id: UUID | str | None = None
    display_name: str
    email: str | None = None
    employee_no: str | None = None
    avatar_url: str | None = None
    status: str


class MemberCreate(BaseModel):
    display_name: str = Field(min_length=1, max_length=120)
    email: str | None = Field(default=None, max_length=255)
    employee_no: str | None = Field(default=None, max_length=80)
    department_id: UUID | None = None


class MemberUpdate(BaseModel):
    display_name: str | None = Field(default=None, min_length=1, max_length=120)
    email: str | None = Field(default=None, max_length=255)
    employee_no: str | None = Field(default=None, max_length=80)
    department_id: UUID | None = None
    status: str | None = None


class CurrentPrincipalRead(BaseModel):
    user: UserRead
    member: MemberRead
    workspace: WorkspaceRead
    permissions: list[str]
    roles: list[str]


class LdapLoginRequest(BaseModel):
    username: str
    password: str


class AuthProviderConfigRead(BaseModel):
    id: str
    workspace_id: str
    provider_type: str
    name: str
    enabled: bool
    config_public: dict[str, object]
    bind_password_configured: bool = False


class AuthProviderConfigUpdate(BaseModel):
    name: str | None = Field(default=None, min_length=1, max_length=120)
    enabled: bool | None = None
    config_public: dict[str, object] | None = None
    bind_password: str | None = None


class AuthProviderTestResult(BaseModel):
    ok: bool
    status: str
    message: str


class TokenRead(BaseModel):
    access_token: str
    token_type: str = "bearer"
